physical/mssql/mssql.go

```diff
@@ -13,10 +13,11 @@ import (
 	"strings"
 	"time"
 
-	metrics "github.com/armon/go-metrics"
-	_ "github.com/denisenkom/go-mssqldb"
+	"github.com/armon/go-metrics"
 	log "github.com/hashicorp/go-hclog"
 	"github.com/hashicorp/go-secure-stdlib/strutil"
+	_ "github.com/microsoft/go-mssqldb"
```

Comment:
I can remove this if folks think this is no longer necessary. Initially we had updated this so that it does not conflict with our MSSQL driver registration for IAM. Since we no longer need MSSQL driver registration in the IAM case, this change is not 'required' for this PR. However, I opted to leave this in because the previous package `"github.com/denisenkom/go-mssqldb"` is out-of-date, and it is even recommended to switch to the official Microsoft package on the old package's [README](https://github.com/denisenkom/go-mssqldb#a-pure-go-mssql-driver-for-gos-databasesql-package)

Reply:
Is this new dependency is a drop-in replacement? I believe Vault Core owns the storage backends code. Maybe we should check with them first? Do you know if our e2e tests are exercising these storage code paths for mssql?

Reply:
Honestly, I did not do the aforementioned due diligence. I think I'm leaning towards removing the changes just because they aren't really necessary/required for this PR, and we can always circle back and update this package when the older one gets deprecated